Packs/CircleCI/Integrations/CircleCI/CircleCI.py

```diff
@@ -51,6 +59,19 @@ def get_workflow_jobs(self, workflow_id: str, page_token: Optional[str] = None):
             params={'page-token': page_token} if page_token else None
         )
 
+    def trigger_workflow(self, vc_type: str, organization: str, project: str, parameters: str):
+        headers = self._headers
+        headers['Circle-Token'] = demisto.params().get('api_key')
```

Comment:
why do you need to add it here? the headers already include the API key under the `Authentication` header.
in any case, you shouldn't use `demisto.params` here, the value is already passed to the client.

Reply:
The headers include the API key as basic authentication, when I tried to call the command with this headers it failed with this error:
`Failed to execute circleci-trigger-workflow command.
Error:
Error in API call [403] - Forbidden
{"message": "Permission denied"}`

for calling the trigger a workflow request we need to use [api_key_header](https://circleci.com/docs/api/v2/#section/Authentication/api_key_header).

I changed the code to save the API key when initialize the client and use it inside `trigger_workflow` method.

For more information:
https://support.circleci.com/hc/en-us/articles/360050351292-How-to-trigger-a-workflow-via-CircleCI-API-v2